### PR TITLE
Ruby on Rails capstone project - Budget app

### PR DESCRIPTION
**Pull request by quavo19**
In this milestone, I @quavo19 implemented the following;

Use ruby gems as software packages system.
Install Ruby on Rails framework.
Understand Rails RESTful design and router.
Use controllers to handle requests and render empty views.
Use params from browser request in a safe way.
Use preprocessed html file with embedded Ruby code.
Use layouts and templates for shared content.
Use database migration files to maintain database schema.
Use validations for models.
Secure app from n+1 problems.
Understand what ORM is.
Write SQL queries with ActiveRecord.
Set up associations between models.
Build a webapp that requires the user to log in.
Use devise gem for authentication.
Limit access to webapp resources based on authorization rules.
Analyze in writing why you have made a coding choice using one structure over another

### CODE REVIEW

File: app/views/categories/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<%= render partial: 'shared/nav', locals: {title:'NEW CATEGORY', symbol:'←', url: :back}%>
+
+<%= render "form", category: @category %>
+
+<br>
+
+<div>
+  <%= link_to "Back to categories", categories_path %>
+</div>
```
Comment (thecodechaser) [inline app/views/categories/new.html.erb:9]: - [ ] Please make sure when the user clicks the "Save" button to create the new category and user should be redirected to the home page(categories page) on success. Currently, it's redirected to the transactions page of that category.

image

File: app/views/expenditures/index.html.erb [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+      <strong>Amount:</strong> <%= expense.amount %>
+    </li>
+  <% end %>
+</ul>
+
+
+
+<div class="container-make-category">
+<%= link_to 'Create Expense', new_category_expenditure_path(@category), class: "make-category"  %>
+</div>
```
Comment (thecodechaser) [inline app/views/expenditures/index.html.erb:17]: - [ ] Please make sure the transitions are presented, and ordered by the most recent as requested in the project requirements. Currently, presented, and ordered by the most last.

image

**Review by thecodechaser — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ 

## Hi @quavo19👋, 

### You are doing a marvelous job on the RoR capstone project and this is one of the best projects I have seen so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
- Nice UI design. ✔️
- No linter errors. :heavy_check_mark:
- Gitflow is followed. :heavy_check_mark:
- Highly organized code. ✔️
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️



## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/expenditures/index.html.erb (hunk 2 above)
Comment (quavo19) [reply app/views/expenditures/index.html.erb:17]: alright sir, all will be done

**Review by omar-muhamad — APPROVED** [review]
# Approved ✔
### Hi @quavo19,

#### Great job, Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

### Congratulations! 🎉 

giphy (1)

## (Highlights) Good Points: :+1:
- Your work is well documented. ✔
- The correct workflow is used.  ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- N/A

## [Optional] suggestions:

- N/A

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Pull request merged**